mypy/dmypy_util.py

```diff
@@ -40,19 +41,76 @@ def send(connection: IPCBase, data: Any) -> None:
     connection.write(json.dumps(data))
 
 
-class WriteToConn:
+class WriteToConn(TextIO):
     """Helper class to write to a connection instead of standard output."""
 
-    def __init__(self, server: IPCBase, output_key: str = "stdout") -> None:
+    def __init__(self, server: IPCBase, output_key: str, isatty: bool) -> None:
         self.server = server
         self.output_key = output_key
+        self._isatty = isatty
+
+    def __enter__(self) -> TextIO:
+        return self
+
+    def __exit__(
+        self,
+        t: type[BaseException] | None,
+        value: BaseException | None,
+        traceback: TracebackType | None,
+    ) -> None:
+        pass
+
+    def __iter__(self) -> Iterator[str]:
+        raise NotImplementedError
+
+    def __next__(self) -> str:
+        raise NotImplementedError
+
+    def close(self) -> None:
+        pass
+
+    def fileno(self) -> int:
+        raise OSError
+
+    def flush(self) -> None:
+        pass
+
+    def isatty(self) -> bool:
+        return self._isatty
+
+    def read(self, n: int = 0) -> str:
+        raise NotImplementedError
```

Comment:
Question: Would it be preferable for these to use https://docs.python.org/3/library/io.html#io.UnsupportedOperation instead?

Reply:
I decided that it probably was, so I've rebased and fixed this.